pkg/migrate/migrate.go

```diff
@@ -134,6 +192,82 @@ func Migrate(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 	return nil
 }
 
+// NewPVMigrator returns a PV migration context
+func NewPVMigrator(cfg *rest.Config, log *log.Logger, srcSC, dstSC string, podReadyTimeout int) (*PVMigrator, error) {
```

Comment:
What is this used by?
What is the use case?

Aren't you initializing the struct directly elsewhere?

Reply:
IMO this whole function is unused and should not exist

I think there should be a single function to run the whole 'can I migrate from sc A to B without having broken access modes' check, but it should take all the args it needs directly.

Reply:
`NewPVMigrator()` is used by kURL's `pvmigrator` CLI: https://github.com/replicatedhq/kURL/pull/3767/files#diff-fc4b9380946b46363bac9e7ecee925d5142b5fa0f607de99b9f5d3383d952e01R97

Reply:
Truthfully what I'm getting from that is 'oh dear lord why did we start reimplementing this logic in kURL, that was the entire reason that the CLI function exists'

I still think the solution is to move the entire 'can I make proper dest PVCs' logic inside of 'Migrate'